icon.c

```diff
@@ -20,6 +21,15 @@
 #include <gdk-pixbuf/gdk-pixbuf.h>
 #include "cairo-pixbuf.h"
 
+static bool bad_iconname(const char* icon_name) {
+	if (strlen(icon_name) > 30) {
```

Comment:
I'd prefer to pick a much higher limit here, e.g. 1024.

Reply:
ok, do as your suggestion